pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/DbLoadedSamlClientConfiguration.java

```diff
@@ -0,0 +1,211 @@
+package org.pac4j.saml.dbclient;
+
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.pac4j.saml.client.SAML2ClientConfiguration;
+import org.pac4j.saml.storage.EmptyStorageFactory;
+import org.pac4j.saml.storage.SAMLMessageStorageFactory;
+
+
+/**
+ * An alternative to {@link SAML2ClientConfiguration}. Unlike {@link SAML2ClientConfiguration}, this class does not use paths to resources.
+ * Instead, it contains the data directly.
+ *
+ * Differences from {@link SAML2ClientConfiguration}:
+ * <ul>
+ * <li>The JKS keystore - contained as a byte array, not a path to a resource</li>
+ * <li>IdP metadata - contained as a string, not a path to a resource</li>
+ * <li>SP metadata - dropped completely</li>
+ * <li>Client name - added</li>
+ * </ul>
+ *
+ * TODO: It would be nice to have a common interface and maybe a common abstract parent too.
+ * 
+ * @author jkacer
+ */
+public class DbLoadedSamlClientConfiguration implements Cloneable {

```

Comment:
Done. I have introduced an abstract parent for both SAML configurations, both the classic one and the database-loaded one extend the abstract parent. This allowed me to get rid of DbLoadedSamlClient completely - hurray.
The configuration is initialized in iternalInit(), as you suggested. But there is one little detail about it to be noted: In the Clients class (in DbLoadedSamlClients now), we must fetch at least all existing client names - we must programatically create the clients and their (empty) configurations and add them to the list. So touching the DB is not completely deferred until configuration initialization. Without this, Clients.findClient(name) would not work because it would not simply know about a specific client at all.
Can you please have a look?